internal/controller/dnsrecord_controller.go

```diff
@@ -358,14 +382,53 @@ func (r *DNSRecordReconciler) SetupWithManager(mgr ctrl.Manager, maxRequeue, val
 			}
 			return toReconcile
 		})).
+		Watches(&v1alpha1.DNSHealthCheckProbe{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+			logger := log.FromContext(ctx)
+			probe, ok := o.(*v1alpha1.DNSHealthCheckProbe)
+			if !ok {
+				logger.V(1).Info("unexpected object type", "error", fmt.Sprintf("%T is not a *v1alpha1.DNSHealthCheckProbe", o))
+				return []reconcile.Request{}
+			}
+			onCluster := &v1alpha1.DNSHealthCheckProbe{}
+			if err := mgr.GetClient().Get(ctx, client.ObjectKeyFromObject(probe), onCluster); client.IgnoreNotFound(err) != nil {
+				logger.Error(err, "failed to get probe")
+				return []reconcile.Request{}
+			}
+
+			probeOwner := &v1alpha1.DNSRecord{
+				ObjectMeta: metav1.ObjectMeta{},
+			}
+			for _, ro := range probe.GetOwnerReferences() {
+				if ro.Kind == "DNSRecord" {
+					probeOwner.Name = ro.Name
+					probeOwner.Namespace = probe.Namespace
+					break
+				}
+			}
+
+			// haven't probed yet or deleting - nothing to do
+			if probe.Status.Healthy == nil {
+				return []reconcile.Request{}
+			}
+
+			// probe.Status.Healthy is not nil here
+			// we probed for the first time - reconcile
+			// or none are nil and status changed - reconcile
+			if onCluster.Status.Healthy == nil || *probe.Status.Healthy != *onCluster.Status.Healthy {
+				return []reconcile.Request{{NamespacedName: client.ObjectKeyFromObject(probeOwner)}}
+			}
+
```

Comment:
There are 2 filters for the reconciliation. This is the first one and will let through only if we flipped ready `Healthy` or did the first probe. The second will list all probes and will do a full loop only if it will result in a change of the status